### PR TITLE
Fix IRsensor with MMU

### PR DESCRIPTION
**Pull request by 3d-gussner**
saves 38bytes

### CODE REVIEW

File: Firmware/ultralcd.cpp [hunk 1]
```diff
@@ -6679,13 +6677,10 @@ static bool selftest_irsensor()
...
-        progress = lcd_selftest_screen(TestScreen::Fsensor, 0, 1, true, 0);
-    }
-    progress = lcd_selftest_screen(TestScreen::Fsensor, progress, 1, true, 0);
-    MMU2::mmu2.unload();
+    progress = lcd_selftest_screen(TestScreen::Fsensor, 0, 1, true, 0);
+    MMU2::mmu2.tool_change(0); //Allow cold extrusion only loads to the gears not nozzle
+    MMU2::mmu2.unload(); //Unload filament
+    set_extrude_min_temp(EXTRUDE_MINTEMP);
 
     for(uint_least8_t i = 0; i < 200; ++i)
```
Comment (gudnimg) [inline Firmware/ultralcd.cpp]: I feel the code below should be deleted as it is part of the old ancient MMU2 test.

File: Firmware/ultralcd.cpp [hunk 2]
```diff
@@ -6669,6 +6666,7 @@ static bool lcd_selftest_fsensor(void)
 //! @retval false failed
 static bool selftest_irsensor()
 {
+    set_extrude_min_temp(0);
     class TempBackup
```
Comment (gudnimg) [inline Firmware/ultralcd.cpp]: Please remove this class. We are no longer heating the extruder to some temperature.

**Review by gudnimg — CHANGES_REQUESTED** [review]
Above are two comments. I feel we should delete the old code since we should really be re-writing the test for the new MMU implementation. The old code is for MMU FW 1.0.6 which we don't support anymore.


Changing `selftest_irsensor()` to this saves 68 bytes:

image

**Comment by gudnimg** [comment]
Here's an easy way to ask the user to select which slot to use for the test:

LCD updates are disabled so the slot selection menu doesn't timeout 👍 

```cpp
static bool selftest_irsensor()
{
    // Ask user which slot to load filament from
    uint8_t slot = choose_menu_P(_T(MSG_SELECT_FILAMENT), _T(MSG_FILAMENT));

    // Render self-test screen
    lcd_selftest_screen(TestScreen::Fsensor, 0, 1, true, 0);

    // Run self-test
    set_extrude_min_temp(0);
    MMU2::mmu2.tool_change(slot);
    MMU2::mmu2.unload(); //Unload filament
    set_extrude_min_temp(EXTRUDE_MINTEMP);
    return true;
}
```

Tested on MK3S+

**Comment by 3d-gussner** [comment]
@gudnimg Please review the PR again. Thanks

**Review by gudnimg — APPROVED** [review]
Looks good to me👍

**Pull request merged**